test/unit/spreadElement.spec.ts

```diff
@@ -47,3 +47,23 @@ test("Spread Element Lua JIT", () => {
     const lua = util.transpileString(`[...[0, 1, 2]]`, options);
     expect(lua).toBe("local ____ = {unpack({\n    0,\n    1,\n    2,\n})}");
 });
+
+test("Spread Element Iterable", () => {
+    const code = `
+        const it = {
+            i: 10,
+            [Symbol.iterator]() {
+                return this;
+            },
+            next() {
+                --this.i;
+                return {
+                    value: this.i,
+                    done: this.i < 0,
+                }
+            }
+        };
+        const arr = [...it];
+        return JSONStringify(arr)`;
+    expect(util.transpileAndExecute(code)).toBe(JSON.stringify([9, 8, 7, 6, 5, 4, 3, 2, 1, 0]));
```

Comment:
I think the other way around would be a bit nicer:
```ts
const result = util.transpileAndExecute(code);
expect(JSON.parse(result)).toEqual([9, 8, 7, 6, 5, 4, 3, 2, 1, 0]);
```

Reply:
I usually do this on purpose to ensure we're not somehow getting array indicies instead of values.